File: flutter_app/lib/features/orders/models/order.dart
class CreateOrderRequest {
  final List<OrderItemRequest> items;
  final int subtotal;
  final int discountAmount;
  final int taxAmount;
  final int total;
  final String paymentMethod;
  final int amountPaid;
  final int changeAmount;
  final String? customerId;
  final String? notes;
  final String outletId;

  CreateOrderRequest({
    required this.items,
    required this.subtotal,
    required this.discountAmount,
    required this.taxAmount,
    required this.total,
    required this.paymentMethod,
    required this.amountPaid,
    required this.changeAmount,
    this.customerId,
    this.notes,
    required this.outletId,
  });

  Map<String, dynamic> toJson() => {
        'items': items.map((i) => i.toJson()).toList(),
        'subtotal': subtotal,
        'discount_amount': discountAmount,
        'tax_amount': taxAmount,
        'total': total,
        'payment_method': paymentMethod,
        'amount_paid': amountPaid,
        'change_amount': changeAmount,
        if (customerId != null) 'customer_id': customerId,
        if (notes != null && notes!.isNotEmpty) 'notes': notes,
        'outlet_id': outletId,
      };
}

class OrderItemRequest {
  final String productId;
  final String variantId;
  final String productName;
  final String variantSize;
  final int price;
  final int quantity;
  final int subtotal;

  OrderItemRequest({
    required this.productId,
    required this.variantId,
    required this.productName,
    required this.variantSize,
    required this.price,
    required this.quantity,
    required this.subtotal,
  });

  Map<String, dynamic> toJson() => {
        'product_id': productId,
        'variant_id': variantId,
        'product_name': productName,
        'variant_size': variantSize,
        'price': price,
        'quantity': quantity,
        'subtotal': subtotal,
      };
}

File: backend/internal/models/order.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Order struct {
	ID              uuid.UUID   `json:"id"`
	OrderNumber     string      `json:"order_number"`
	OutletID        uuid.UUID   `json:"outlet_id"`
	CashierID       uuid.UUID   `json:"cashier_id"`
	CashierName     string      `json:"cashier_name"`
	CustomerID      *uuid.UUID  `json:"customer_id,omitempty"`
	CustomerName    *string     `json:"customer_name,omitempty"`
	Items           []OrderItem `json:"items"`
	Subtotal        int64       `json:"subtotal"`
	DiscountAmount  int64       `json:"discount_amount"`
	TaxAmount       int64       `json:"tax_amount"`
	SurchargeAmount int64       `json:"surcharge_amount"`
	Total           int64       `json:"total"`
	PaymentMethod   string      `json:"payment_method"`
	AmountPaid      int64       `json:"amount_paid"`
	ChangeAmount    int64       `json:"change_amount"`
	Status          string      `json:"status"`
	Notes           *string     `json:"notes,omitempty"`
	RefundReason    *string     `json:"refund_reason,omitempty"`
	RefundedAt      *time.Time  `json:"refunded_at,omitempty"`
	CreatedAt       time.Time   `json:"created_at"`
}

type OrderItem struct {
	ID          uuid.UUID `json:"id"`
	OrderID     uuid.UUID `json:"order_id"`
	ProductID   uuid.UUID `json:"product_id"`
	VariantID   uuid.UUID `json:"variant_id"`
	ProductName string    `json:"product_name"`
	VariantSize string    `json:"variant_size"`
	Price       int64     `json:"price"`
	Quantity    int       `json:"quantity"`
	Subtotal    int64     `json:"subtotal"`
}

type CreateOrderRequest struct {
	Items          []CreateOrderItemRequest `json:"items"`
	Subtotal       int64                    `json:"subtotal"`
	DiscountAmount int64                    `json:"discount_amount"`
	TaxAmount      int64                    `json:"tax_amount"`
	Total          int64                    `json:"total"`
	PaymentMethod  string                   `json:"payment_method"`
	AmountPaid     int64                    `json:"amount_paid"`
	ChangeAmount   int64                    `json:"change_amount"`
	CustomerID     *string                  `json:"customer_id,omitempty"`
	Notes          *string                  `json:"notes,omitempty"`
	OutletID       string                   `json:"outlet_id"`
}

type CreateOrderItemRequest struct {
	ProductID   string `json:"product_id"`
	VariantID   string `json:"variant_id"`
	ProductName string `json:"product_name"`
	VariantSize string `json:"variant_size"`
	Price       int64  `json:"price"`
	Quantity    int    `json:"quantity"`
	Subtotal    int64  `json:"subtotal"`
}
